src/rvsdg/egglog_optimizer/mod.rs

```diff
@@ -0,0 +1,63 @@
+use bril_rs::Type;
+
+use self::constant_fold::constant_fold_egglog;
+
+pub(crate) mod constant_fold;
+
+pub fn rvsdg_egglog_code() -> String {
+    let code = vec![
+        include_str!("schema.egg").to_string(),
+        include_str!("subst.egg").to_string(),
+        constant_fold_egglog(),
+    ];
+    code.join("\n")
+}
+
+pub fn rvsdg_egglog_schedule() -> String {
+    "(run 3)".to_string()
+}
+
+#[derive(Debug, PartialEq, Clone)]
+struct BrilOp {
+    op: &'static str,
+    egglog_op: &'static str,
+    // so far we don't need more than 2 types
+    // if the field is None, it is beyond the number of inputs
+    input_types: [Option<Type>; 2],
```

Comment:
this looks a bit uncommon to me. Why not `Vec<Type>`?

Reply:
I'm declaring a global type so it needs to have a fixed size apparently